ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java

```diff
@@ -126,14 +125,22 @@
 
     @Override
     public Job recoverWriteJob(final UUID jobId) throws SignatureException, IOException, XmlProcessingException, JobRecoveryException {
-        // TODO - come back and re-implement this
-        throw new UnsupportedOperationException();
+        ModifyJobResponse jobResponse = this.client.modifyJob(new ModifyJobRequest(jobId));
+        if (RequestType.PUT != jobResponse.getMasterObjectList().getRequestType()){
+            throw new JobRecoveryException("PUT", jobResponse.getMasterObjectList().getRequestType().toString() );
+        }
+
+        return new WriteJobImpl(this.client, jobResponse.getMasterObjectList());
     }
 
     @Override
     public Job recoverReadJob(final UUID jobId) throws SignatureException, IOException, XmlProcessingException, JobRecoveryException {
-        // TODO - come back and re-implement this
-        throw new UnsupportedOperationException();
+        ModifyJobResponse jobResponse = this.client.modifyJob(new ModifyJobRequest(jobId));
+        if (RequestType.GET != jobResponse.getMasterObjectList().getRequestType()){
+            throw new JobRecoveryException("GET", jobResponse.getMasterObjectList().getRequestType().toString() );
```

Comment:
Instead of typing "GET" I would do the following. `RequestType.GET.toString()` or something like that.